app/src/main/java/com/amaze/filemanager/asynchronous/management/ServiceWatcherUtil.java

```diff
@@ -0,0 +1,253 @@
+package com.amaze.filemanager.asynchronous.management;
+
+/**
+ * Created by vishal on 4/1/17.
+ *
+ * Helper class providing helper methods to manage Service startup and it's progress
+ * Be advised - this class can only handle progress with one object at a time. Hence, class also provides
+ * convenience methods to serialize the service startup.
+ */
+
+import android.app.NotificationManager;
+import android.content.Context;
+import android.content.Intent;
+import android.support.v4.app.NotificationCompat;
+import android.text.format.Formatter;
+
+import com.amaze.filemanager.R;
+import com.amaze.filemanager.asynchronous.AbstractRepeatingRunnable;
+import com.amaze.filemanager.asynchronous.services.AbstractProgressiveService;
+import com.amaze.filemanager.ui.notifications.NotificationConstants;
+import com.amaze.filemanager.utils.ProgressHandler;
+
+import java.lang.ref.WeakReference;
+import java.util.concurrent.*;
+
+import static com.amaze.filemanager.asynchronous.management.ServiceWatcherUtil.ServiceStatusCallbacks.*;
+import static com.amaze.filemanager.asynchronous.management.ServiceWatcherUtil.ServiceStatusCallbacks.STATE_UNSET;
```

Comment:
Redundant import.

Reply:
We should turn on lint for automated redundant imports checks... Will fix these ones.

Reply:
Fixed. But, was there any end user perf effect?